packages/react-server/src/ReactFlightServer.js

```diff
@@ -320,6 +340,33 @@ function serializeThenable(request: Request, thenable: Thenable<any>): number {
   return newTask.id;
 }
 
+export function emitHint(code: string, model: HintModel): void {
+  const request = resolveCurrentRequest();
```

Comment:
It's a little unfortunate that we have to do this twice. I don't really trust that `get()` to be fast, and this is just a layering issue of our own making.

We can expose `resolveRequest()` and then pass that in as an argument to `emitHint`. The Request is already exposed, it's just considered opaque to the consumer.

Then we can make `resolveHints()` be `getHints(request)` just like `getRoot(response)` on the client.

```
const req = resolveCurrentRequest();
const hints = getHints(req);
// Check for duplicates
emitHint(req, code, model);
```